glnx-fdio.c

```diff
@@ -741,6 +724,37 @@ glnx_regfile_copy_bytes (int fdf, int fdt, off_t max_bytes, gboolean try_reflink
             m = (size_t) max_bytes;
         }
 
+      /* First try copy_file_range(), unless we already tried.  Note this is an
+       * inlined version of try_copy_file_range() from systemd upstream, which
+       * works better since we use POSIX errno style.
+       */
+      if (try_cfr)
+        {
+          n = copy_file_range (fdf, NULL, fdt, NULL, m, 0u);
+          if (n < 0)
+            {
+              if (errno == ENOSYS)
+                {
+                  have_cfr = 0;
+                  try_cfr = false;
```

Comment:
I think I would prefer doing the `ENOSYS` discovery ahead of time, rather than making the loop logic more complicated. E.g. replacing the `if (try_cfr)` with `if (can_copy_file_range ())` where the static var lives there. Though it's up to you!